### PR TITLE
[wip] spotify

### PR DESCRIPTION
**Pull request by hf**
## What kind of change does this PR introduce?

Bug fix, feature, docs update, ...

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: internal/models/linking_test.go [hunk 1]
```diff
@@ -82,18 +82,6 @@ func (ts *AccountLinkingTestSuite) TestCreateAccountDecisionWithAccounts() {
...
-	decision, err = DetermineAccountLinking(ts.db, "sso:f06f9e3d-ff92-4c47-a179-7acf1fda6387", "abcdefgh", []string{"other@samltest.id"})
-	require.NoError(ts.T(), err)
-
-	require.Equal(ts.T(), decision.Decision, CreateAccount)
-
-	// when looking for an email that doesn't exist in the SSO linking domain
-	decision, err = DetermineAccountLinking(ts.db, "sso:f06f9e3d-ff92-4c47-a179-7acf1fda6387", "abcdefgh", []string{"text@example.com"})
-	require.NoError(ts.T(), err)
-
-	require.Equal(ts.T(), decision.Decision, CreateAccount)
```
Comment (hf) [inline internal/models/linking_test.go:96]: Just wandering why these were deleted? Seems unrelated to the changes.

Comment (kangmingtay) [reply internal/models/linking_test.go:96]: i was also testing the account linking behaviour and noticed that these tests were repeated so i decided to remove them
